src/poker-evaluator.ts

```diff
@@ -64,11 +66,61 @@ function evaluate(cardValues: number[]): EvaluatedHand {
   if (cardValues.length === 5 || cardValues.length === 6) {
     p = evalCard(p);
   }
+  const handType = (p === ROYAL_FLUSH_SCORE) ? 10 : p >> 12;
 
   return {
-    handType: p >> 12,
+    handType: handType,
     handRank: p & 0x00000fff,
     value: p,
-    handName: HAND_TYPES[p >> 12]
+    handName: HAND_TYPES[handType]
+  }
+}
+
+export function winningOdds (hand: string[], community: string[], playerCount: number, cycles: number): number {
+  // Above 23 players, we run out of cards in a deck.  23 * 2 + 5 = 51
+  if (playerCount > 23) {
+    throw new Error("You may have at most 23 players.")
+  }
+
+  const numHand = convertCardsToNumbers(hand)
+  const numCommunity = convertCardsToNumbers(community)
+  const startingDeck = deckWithoutSpecifiedCards([...numHand, ...numCommunity]);
+  let wins = 0;
+  for (let i = 0 ; i < cycles ; i ++ ) {
```

Comment:
Stylistic nit: please remove extra spaces in all for loops. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for for stylistic conventions.